src/hosted-form/iframe-content/hosted-input-validator.ts

```diff
@@ -1,39 +1,60 @@
 import { cvv, expirationDate, number } from 'card-validator';
-import { pick } from 'lodash';
+import { getTypeInfo } from 'credit-card-type';
+import { includes } from 'lodash';
 import { object, string, StringSchema, ValidationError } from 'yup';
 
+import { CardInstrument } from '../../payment/instrument';
+import HostedFieldType from '../hosted-field-type';
+
 import { HostedInputValidateErrorDataMap } from './hosted-input-validate-error-data';
 import HostedInputValidateResults from './hosted-input-validate-results';
 import HostedInputValues from './hosted-input-values';
 
-interface HostedInputValidateOptions {
-    isCardCodeRequired?: boolean;
-}
-
 export default class HostedInputValidator {
-    async validate(values: HostedInputValues, options?: HostedInputValidateOptions): Promise<HostedInputValidateResults> {
-        const schemas: { [key in keyof HostedInputValues]: StringSchema } = {
-            cardExpiry: this._getCardExpirySchema(),
-            cardName: this._getCardNameSchema(),
-            cardNumber: this._getCardNumberSchema(),
-        };
+    constructor(
+        private _cardInstrument?: CardInstrument
+    ) {}
 
+    async validate(values: HostedInputValues, fields?: HostedFieldType[]): Promise<HostedInputValidateResults> {
+        const requiredFields = fields || Object.keys(values);
```

Comment:
shouldn't you always be passing the fields? What's the scenario where you get the fields based on the values?

Reply:
I guess I could make `fields` a required parameter, or infer the field types based on the `values`.

Reply:
yeah what I was thinking is that if you rely on values only, you could potentially miss to pass a value and validate the schema based on the values itself. I feel it would be easier to either stick to always relying on values, or making `fields` mandatory